app/lib/utils/sanitize.js

```diff
@@ -0,0 +1,17 @@
+import truncate from './truncate'
+
+function isObject(x) {
+  return x != null && typeof x === 'object'
+}
+
+const sanitize = data => {
+  if (isObject(data)) {
+    const sanitizedData = { ...data }
+    const props = ['lndconnectUri', 'lndconnectQRCode']
+    props.forEach(key => {
+      sanitizedData[key] = truncate(data[key])
+    })
+  }
```

Comment:
To me looks like this function doesn't actually do anything. I'd assume it meant to return `sanitizedData`. If that is the case another comment would be that `const sanitizedData = { ...data }` will destroy prototype data so in cases like with `lndConfig` instances it can lead to all sorts of obscure bugs

Reply:
I've updated this to return `sanitizedData`. The strange thing though was that it was working as expected without.

> If that is the case another comment would be that `const sanitizedData = { ...data }` will destroy prototype data so in cases like with `lndConfig` instances it can lead to all sorts of obscure bugs

This primary purpose of this method is to allow us to prevent sensitive data from being logged on the terminal, such as when we are logging the content of an `LndConfig` object.

`const sanitizedData = { ...data }` ensure that we modify a clone of the original object rather than modifying the original object directly, which we don't want to do.

We aren't intending to be left with a functional `LndConfig` object after running through this method.

Reply:
> ensure that we modify a clone of the original object

That's not the question. The intent is obvious. Was just pointing out that there exists a side effect which removes a proto. If, as you say, this is util has a very narrow scope of use it'd be good to provide an extensive description/comment pointing out that specific, since it's placed in `app/lib/utils/sanitize` which is a top level util dir and those utils it can potentially be widely used across the app. Or another option is to move it somewhere closer to where it's actually being used.

Reply:
I've added some documentation to this method and included a note about the potential unintended side effects.